File: daily_python/3068_Find_the_Maximum_Sum_of_Node_Values.py
import os
import sys
from typing import List

class Solution:
    def maximumValueSum(self, nums: List[int], k: int, edges: List[List[int]]) -> int:
        ans = 0
        odd = 0
        m = float('inf')
        for temp in nums:
            ori = temp
            trans = temp ^ k
            if trans > ori:
                odd ^= 1
            m = min(m, max(ori, trans) - min(ori, trans))
            ans += max(ori, trans)
        if odd:
            ans -= m
        return ans

def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            nums = list(map(int, lines[0].split('=')[1].strip()[1:-2].split(',')))
            k = int(lines[1].split('=')[1].strip()[:-1])
            edges = [list(map(int, edge.split(','))) for edge in lines[2].split('=')[1].strip()[2:-3].split('},{')]
            dataset.append((nums, k, edges))
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for nums, k, edges in dataset:
        results.append(solution.maximumValueSum(nums, k, edges))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')

File: daily/2025/3068_Find_the_Maximum_Sum_of_Node_Values.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    long long maximumValueSum(vector<int>& nums, int k, vector<vector<int>>& edges) {
        long long ans = 0;
        bool changed_odd = false;
        int minimum_diff = INT_MAX;
        for (auto temp : nums){
            int original = temp;
            int transfer = temp ^ k;
            int diff = max(original, transfer) - min(original, transfer);
            minimum_diff = min(minimum_diff, diff);
            if (transfer > original) changed_odd = !changed_odd;
            ans += max(original, transfer);
        }
        if (changed_odd) ans -= minimum_diff;
        return ans;
    }
};

int main(){
    vector<int> nums = {1,2,1};
    int k = 3;
    vector<vector<int>> edges = {{0,1},{0,2}};
    // vector<int> nums = {2,3};
    // int k = 7;
    // vector<vector<int>> edges = {{0,1}};
    // vector<int> nums = {7,7,7,7,7,7};
    // int k = 3;
    // vector<vector<int>> edges = {{0,1},{0,2},{0,3},{0,4},{0,5}};
    // vector<int> nums = {24,78,1,97,44};
    // int k = 6;
    // vector<vector<int>> edges = {{0,2},{1,2},{4,2},{3,4}};
    Solution S;
    cout << S.maximumValueSum(nums, k, edges) << endl;
    return 0;
}
